From ac4fdb2848310607d5d3209f3882431e63b7c6c1 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Date: Tue, 25 Mar 2025 22:32:46 +0100
Subject: [PATCH 1665/1695] drivers: stm32_risaf: refactor granularity getting
 from hw config

_RISAF_HWCFGR content is already stored in a local variable.
Formula is simple enough not to use a temporary local variable.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 33d63e4f7b69f3990b8cfc153f1b9a9b2b15f919 drivers: stm32_risaf: refactor granularity getting from hw config]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Change-Id: I8068f91c86d919441d1f7b94703799b475d9623f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/443825
Tested-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/firewall/stm32_risaf.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/core/drivers/firewall/stm32_risaf.c b/core/drivers/firewall/stm32_risaf.c
index 4474ff65e..1317a1dce 100644
--- a/core/drivers/firewall/stm32_risaf.c
+++ b/core/drivers/firewall/stm32_risaf.c
@@ -458,7 +458,6 @@ void stm32_risaf_is_locked(struct stm32_risaf_instance *risaf, bool *state)
 static TEE_Result stm32_risaf_init_ddata(struct stm32_risaf_instance *risaf)
 {
 	vaddr_t base = risaf_base(risaf);
-	uint32_t granularity = 0;
 	uint32_t mask_lsb = 0;
 	uint32_t mask_msb = 0;
 	uint32_t hwcfgr = 0;
@@ -478,10 +477,8 @@ static TEE_Result stm32_risaf_init_ddata(struct stm32_risaf_instance *risaf)
 					 _RISAF_HWCFGR_CFG1_SHIFT;
 
 	/* Get IP region granularity */
-	granularity = io_read32(risaf_base(risaf) + _RISAF_HWCFGR);
-	granularity = BIT((granularity & _RISAF_HWCFGR_CFG3_MASK) >>
-			  _RISAF_HWCFGR_CFG3_SHIFT);
-	risaf->ddata->granularity = granularity;
+	risaf->ddata->granularity = BIT((hwcfgr & _RISAF_HWCFGR_CFG3_MASK) >>
+					_RISAF_HWCFGR_CFG3_SHIFT);
 
 	return TEE_SUCCESS;
 }
-- 
2.47.2

